homeassistant/util/async_.py

```diff
@@ -76,3 +74,33 @@ def run_callback() -> None:
 
     loop.call_soon_threadsafe(run_callback)
     return future
+
+
+async def safe_wait(
+    tasks: Any, logger: Optional[logging.Logger] = None, return_exceptions: bool = False
+) -> List[Any]:
+    """Safe version of wait and gather.
+
+    It work like gather but wait don't break any workflows.
+    It allow also to log exception in correct namespace.
+    """
+    tasks = list(tasks)
+
+    if not tasks:
+        return []
+
+    results = await asyncio.gather(*tasks, return_exceptions=True)
```

Comment:
Gather is mostly that code from my origin PR. So you replace a low level function to call a top level function they call a top level function. Gather is a helper around wait and do what my function does (mostly)

Reply:
And call an helper they call an helper is not real performant

Reply:
Let's discuss it when you get back from holiday 👍 

`wait` returns a `set` so order of results was not guaranteed.